### PR TITLE
713-yc-EdgeNumberDistance

### PR DESCRIPTION
**Pull request by c-yuwei**
closes issue #713

### CODE REVIEW

**Review by annacanal — CHANGES_REQUESTED** [review]
You need to update this branch with develop.
braph2 command gives an error in this branch:

Error using cd
Cannot CD to /home/annacg/Documents/Braph-2.0-Matlab/braph2/graph/measures/MEX_SRC (Name is nonexistent or not a directory).

Error in compile_mex (line 22)
    cd(path_files)

Error in braph2 (line 42)
    compile_mex

**Comment by maverickyuwei** [comment]
> You need to update this branch with develop.
> braph2 command gives an error in this branch that will be updated with develop:
> 
> Error using cd
> Cannot CD to /home/annacg/Documents/Braph-2.0-Matlab/braph2/graph/measures/MEX_SRC (Name is nonexistent or not a directory).
> 
> Error in compile_mex (line 22)
> cd(path_files)
> 
> Error in braph2 (line 42)
> compile_mex

Done, b38eecb
Thank you Anna, this up-to-date check will be always verified from now on.

File: braph2/graph/measures/EdgeNumberDistance.m [hunk 1]
```diff
@@ -0,0 +1,153 @@
+classdef EdgeNumberDistance < Distance
+    % EdgeNumberDistance Edge number distance measure
+    % EdgeNumberDistance provides the edge number distance of a graph for 
+    % weighted undirected (WU) and weighted directed (WD) graphs.
+    % It is calculated as the number of edges in the shortest weighted path
+    % between two nodes.
```
Comment (annacanal) [inline braph2/graph/measures/EdgeNumberDistance.m:6]: Add one empty comment line before it is calculated
%
% It is calculated as the number of edges in the shortest weighted path
% between two nodes.

Comment (annacanal) [reply braph2/graph/measures/EdgeNumberDistance.m:6]: also: within the same layer

File: braph2/graph/measures/EdgeNumberDistance.m [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    % See also Measure, Distance, GraphWU, GraphWD, MultiplexGraphWU, MultiplexGraphWD.
+   
+    methods
+        function m = EdgeNumberDistance(g, varargin)
+            % EDGENUMBERDISTANCE(G) creates edge number distance with default properties.
+            % G is a graph (e.g, an instance of GraphBD, GraphBU,
+            % GraphWD, GraphWU, MultiplexGraphBD, MultiplexGraphBU, MultiplexGraphWD
+            % or MultiplexGraphWU). 
+            %
+            % See also Measure, Distance, GraphWU, GraphWD, MultiplexGraphWU, MultiplexGraphWD.
```
Comment (annacanal) [inline braph2/graph/measures/EdgeNumberDistance.m:32]: No GraphBD, GraphBU... not binary

File: braph2/graph/measures/EdgeNumberDistance.m [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+            % GETDESCRIPTION returns the edge number distance description 
+            %
+            % DESCRIPTION = GETDESCRIPTION() returns the description of the
+            % edge number distance measure.
+            %
+            % See also getClass, getName.
+            
+            description = [ ...
+                'The edge distance number of a graph is ' ...
+                'the number of edges in the shortest weighted path between two nodes. ' ...
```
Comment (annacanal) [inline braph2/graph/measures/EdgeNumberDistance.m:81]: between two nodes within a layer

File: braph2/graph/measures/test_EdgeNumberDistance.m [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+known_value = {[
+                0 2 1 1 0;
+                2 0 1 1 0;
+                1 1 0 2 0;
+                1 1 2 0 0;
+                0 0 0 0 0;
+                ]};
+
+g = GraphWU(A);
+edge_number_distance = EdgeNumberDistance(g);;
```
Comment (annacanal) [inline braph2/graph/measures/test_EdgeNumberDistance.m:21]: delete one: ';'

File: braph2/graph/measures/test_EdgeNumberDistance.m [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+known_value = {[
+                0 2 1 1 0;
+                1 0 2 2 0;
+                1 1 0 1 0;
+                1 1 2 0 0;
+                0 0 0 0 0;
+                ]};
+
+g = GraphWD(A);
+edge_number_distance = EdgeNumberDistance(g);;
```
Comment (annacanal) [inline braph2/graph/measures/test_EdgeNumberDistance.m:45]: same here

File: braph2/graph/measures/test_EdgeNumberDistance.m [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+               [
+                0 2 1 1 0;
+                2 0 1 1 0;
+                1 1 0 2 0;
+                1 1 2 0 0;
+                0 0 0 0 0;
+               ]};
+
+g = MultiplexGraphWU(A);
+edge_number_distance = EdgeNumberDistance(g);;
```
Comment (annacanal) [inline braph2/graph/measures/test_EdgeNumberDistance.m:89]: same here

File: braph2/graph/measures/test_EdgeNumberDistance.m [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+               [
+                0 2 1 1 0;
+                1 0 2 2 0;
+                1 1 0 1 0;
+                1 1 2 0 0;
+                0 0 0 0 0;
+               ]};
+
+g = MultiplexGraphWD(A);
+edge_number_distance = EdgeNumberDistance(g);;
```
Comment (annacanal) [inline braph2/graph/measures/test_EdgeNumberDistance.m:133]: same here

**Review by annacanal — CHANGES_REQUESTED** [review]

File: braph2/graph/measures/EdgeNumberDistance.m (hunk 1 above)
Comment (maverickyuwei) [reply braph2/graph/measures/EdgeNumberDistance.m:6]: 45664dd

File: braph2/graph/measures/EdgeNumberDistance.m (hunk 2 above)
Comment (maverickyuwei) [reply braph2/graph/measures/EdgeNumberDistance.m:33]: 32cd066

File: braph2/graph/measures/EdgeNumberDistance.m (hunk 3 above)
Comment (maverickyuwei) [reply braph2/graph/measures/EdgeNumberDistance.m:82]: 8034e79

File: braph2/graph/measures/test_EdgeNumberDistance.m (hunk 7 above)
Comment (maverickyuwei) [reply braph2/graph/measures/test_EdgeNumberDistance.m:133]: 778dbcc

File: braph2/graph/measures/test_EdgeNumberDistance.m (hunk 4 above)
Comment (maverickyuwei) [reply braph2/graph/measures/test_EdgeNumberDistance.m]: 778dbcc

File: braph2/graph/measures/test_EdgeNumberDistance.m (hunk 5 above)
Comment (maverickyuwei) [reply braph2/graph/measures/test_EdgeNumberDistance.m]: 778dbcc

File: braph2/graph/measures/test_EdgeNumberDistance.m (hunk 6 above)
Comment (maverickyuwei) [reply braph2/graph/measures/test_EdgeNumberDistance.m]: 778dbcc

**Review by annacanal — APPROVED** [review]

**Review by giovannivolpe — APPROVED** [review]

**Pull request merged**